File: frontend/kube-guard-frontend/src/api/clusters.ts
// src/api/clusters.ts
import { get, post, del } from './http'

export type ClusterRow = {
  id: number
  name: string
  type: 'self-hosted' | 'managed'
  provider: string
  is_active: number // 0/1
  created_ts: number
  updated_ts: number
}

export type ActiveClusterResp = {
  ok: boolean
  active: ClusterRow | null
}

export type VerifyClusterReq = {
  kubeconfig: string
}

export type VerifyClusterResp = {
  ok: boolean
  test?: {
    ok: string
    server: string
    nodes: string
  }
}

export type AddClusterReq = {
  name: string
  type: 'self-hosted' | 'managed'
  provider: string
  kubeconfig: string
}

export type AddClusterResp = {
  ok: boolean
  id?: number
  test?: any
}

export function fetchClusters() {
  return get<ClusterRow[]>('/api/clusters')
}

export function fetchActiveCluster() {
  return get<ActiveClusterResp>('/api/clusters/active')
}

export function verifyCluster(req: VerifyClusterReq) {
  return post<VerifyClusterResp>('/api/clusters/verify', req)
}

export function addCluster(req: AddClusterReq) {
  return post<AddClusterResp>('/api/clusters', req)
}

export function activateCluster(clusterId: number) {
  return post<{ ok: boolean; active_cluster_id: number; kubeconfig_path?: string }>(
    `/api/clusters/${clusterId}/activate`,
    {}
  )
}

export function deleteCluster(clusterId: number) {
  return del<{ ok: boolean }>(`/api/clusters/${clusterId}`)
}

File: backend/kube-devops-backend/fastApiProject/routers/clusters.py
# routers/clusters.py
from __future__ import annotations

from typing import Literal

from fastapi import APIRouter
from pydantic import BaseModel, Field

from services.clusters.service import (
    add_cluster,
    activate_cluster,
    delete_cluster,
    get_active_cluster,
    list_clusters,
    verify_cluster,
)

router = APIRouter(prefix="/api/clusters", tags=["clusters"])


class AddClusterReq(BaseModel):
    name: str = Field(..., min_length=1)
    type: Literal["self-hosted", "managed"] = "managed"
    provider: str = ""
    kubeconfig: str = Field(..., min_length=1)


class VerifyClusterReq(BaseModel):
    kubeconfig: str = Field(..., min_length=1)


@router.get("")
def api_list_clusters():
    return list_clusters()


# ✅ 新增：当前 active 集群
@router.get("/active")
def api_get_active_cluster():
    return {"ok": True, "active": get_active_cluster()}


# ✅ 新增：只校验 kubeconfig（不入库）
@router.post("/verify")
def api_verify_cluster(req: VerifyClusterReq):
    return verify_cluster(req.kubeconfig)


@router.post("")
def api_add_cluster(req: AddClusterReq):
    return add_cluster(req.name, req.type, req.provider, req.kubeconfig)


@router.post("/{cluster_id}/activate")
def api_activate_cluster(cluster_id: int):
    return activate_cluster(cluster_id)


@router.delete("/{cluster_id}")
def api_delete_cluster(cluster_id: int):
    return delete_cluster(cluster_id)
